From 98b443c4ed6f447c6b7a4dcd012f56e666d7de4a Mon Sep 17 00:00:00 2001
From: Uma Maheswara Rao G <uma.gangumalla@intel.com>
Date: Wed, 10 Feb 2016 14:54:45 -0800
Subject: [PATCH 1212/2848] HADOOP-12788. OpensslAesCtrCryptoCodec should log
 which random number generator is used.
 Contributed by Wei-Chiu Chuang.

(cherry picked from commit df4dcd3e75f1f935bd9a537e63758fee71cf4e02)

Change-Id: Ic874e925310b3b965196c800dadfb1b48777c5e4
---
 .../hadoop/crypto/OpensslAesCtrCryptoCodec.java    |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/OpensslAesCtrCryptoCodec.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/OpensslAesCtrCryptoCodec.java
index 4ca79b3..63c382c 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/OpensslAesCtrCryptoCodec.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/OpensslAesCtrCryptoCodec.java
@@ -61,6 +61,9 @@ public void setConf(Configuration conf) {
         Random.class);
     try {
       random = ReflectionUtils.newInstance(klass, conf);
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Using " + klass.getName() + " as random number generator.");
+      }
     } catch (Exception e) {
       LOG.info("Unable to use " + klass.getName() + ".  Falling back to " +
           "Java SecureRandom.", e);
-- 
1.7.9.5

